tmail-backend/tmail-third-party/open-ai/src/main/java/com/linagora/tmail/mailet/ChatLanguageModelFactory.java

```diff
@@ -0,0 +1,27 @@
+package com.linagora.tmail.mailet;
+
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.openai.OpenAiChatModel;
+
+public class ChatLanguageModelFactory {
+    private static final String OPEN_AI_BASE_URL = "https://api.openai.com/v1";
+    private static final String LINAGORA_AI_BASE_URL = "https://ai.linagora.com/api/";
+
+    public ChatLanguageModel createChatLanguageModel(MailBotConfig config) {
+        String apiKey = config.getApiKey();
+        LlmModel llmModel = config.getLlmModel();
+
+        return switch (llmModel.llm()) {
+            case OPEN_AI -> createOpenAILanguageModel(apiKey, llmModel.modelName(), OPEN_AI_BASE_URL);
+            case LINAGORA_AI -> createOpenAILanguageModel(apiKey, llmModel.modelName(), LINAGORA_AI_BASE_URL);
```

Comment:
To be faily honnest I'd prefer having a `<baseURL>` property in the mailet configuration and not hard code ai.linagra.com URL in the code.

Reply:
Sure, it shouldn't be hard to add. That would allow users to submit their custom OpenAI-based models also, I like the idea.

Reply:
Done. Now, the Llm enum refers to the API type rather than a specific LLM. I'm thinking of renaming it to LlmAPI or LlmType, any suggestions of better names.

Reply:
Not relevant anymore as the LLM enum is deleted.